core/src/test/java/org/axonframework/eventhandling/TrackingEventProcessorTest.java

```diff
@@ -656,6 +656,17 @@ public void testWhenFailureDuringInit() throws InterruptedException {
 
     }
 
```

Comment:
Nit: Although not as important as testing the black listing API call, I think it might be valuable to add a test for the `TrackingEventProcessor#hasAvailableThreads()` function as well.

Reply:
Right! Done.